Use the binary package if there is one (#2)
sixtyfpsui/sixtyfps-cpp-template: CMakeLists.txt

cmake_minimum_required(VERSION 3.19)
project(my_application LANGUAGES CXX)

> 

Reviewer: For use later, to show when the optional package was found:

```suggestion
include(FeatureSummary)
```

> find_package(SixtyFPS)

Author: should we use QUIET here or not?

By default, cmake will output this warning if the package was not found:

```
CMake Warning at CMakeLists.txt:4 (find_package):
  By not providing "FindSixtyFPS.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "SixtyFPS",
  but CMake did not find one.

  Could not find a package configuration file provided by "SixtyFPS" with any
  of the following names:

    SixtyFPSConfig.cmake
    sixtyfps-config.cmake

  Add the installation prefix of "SixtyFPS" to CMAKE_PREFIX_PATH or set
  "SixtyFPS_DIR" to a directory containing one of the above files.  If
  "SixtyFPS" provides a separate development package or SDK, be sure it has
  been installed.
```

So we have three choices
 1. leave it like that
 2. use QUIET,  so it doesn't show any message and just use SixtyFPS from git
 3. use QUIET, but still add a `message("No binary packages of SixtyFPS found, cmake will download and compile packages from git. Checkout https://sixtyfps.io/docs/cpp/install.html for more info")` (or something like that)

Reviewer: Ah, you had the same idea at the same time hehe

Reviewer: My preference is (3).

Ultimately I think what's really important is that the cmake configure output gives an indication what happened / is going to happen. So with (3) it would tell what's going to happen when the package could not be found in the cmake module search path. But perhaps we should also write an explicit message *when* it was found.

> find_package(SixtyFPS)
> if (NOT SixtyFPS_FOUND)
>   # If SixtyFPS packages is not found, download from git
>   include(FetchContent)

Reviewer: I think it would be worthwhile to show a clarifying message, because by default when building not against a package, this will still produce messages that look like something is wrong.

```suggestion
find_package(SixtyFPS QUIET)
if (NOT SixtyFPS_FOUND)
  message("SixtyFPS could not be located in the CMake module search path. Downloading it from Git and building it locally");
  include(FetchContent)
```

Maybe it should be "QUIET" though but still continue to show the "error"?

  FetchContent_Declare(
    SixtyFPS
    GIT_REPOSITORY https://github.com/sixtyfpsui/sixtyfps.git
    GIT_TAG v0.1.4
    SOURCE_SUBDIR api/sixtyfps-cpp
  )
  FetchContent_MakeAvailable(SixtyFPS)

> endif (NOT SixtyFPS_FOUND)

Reviewer: ```suggestion
else ()
    message("Using SixtyFPS package from ${SixtyFPS_DIR}")
endif()
```

Author: Why?  Is it something that is commonly done?
I personally don't like when there is a lot of "noise" in the cmake log. But maybe that's not noise?

Reviewer: Maybe you're right. A regular `find_package` also wouldn't print the summary. Feel free to skip :)
